Outposts v0.2: Test PR

tmtmtl30 opened the pull request:
## About The Pull Request

ships spawn in (non-exclusive, so don't get too comfortable) hangars at the outpost sized to them, connected to a new "main level" of the outpost via an elevator. new hangars are generated as necessary as ships spawn / dock. let me know if radio works across hangars or not

~~this is mega-unfinished and not fit for merge, but it IS fit for test, and that's what we do around here isn't it???~~

i'm basically done with this. that took a hot minute.
for detailed documentation, check here: https://hackmd.io/@Xegx-gsTR2WyYII99ZKtrg/B1updZdK2

## Why It's Good For The Game

it's cool

## Changelog

:cl:
add: Ships now spawn at the outpost, in hangars. Take the elevator in the top-right down to the main level.
add: 100% guaranteed most realistic ss13 elevator experience if you ignore all the things that aren't realistic
/:cl:

tmtmtl30 committed:
hangar prototype

tmtmtl30 committed:
Merge branch 'master' into hangar

tmtmtl30 committed:
elevators, basic outpost spawning (no maps)

tmtmtl30 committed:
Merge branch 'master' into hangar

tmtmtl30 committed:
prototype outpost for test

tmtmtl30 committed:
Merge remote-tracking branch 'origin/master' into hangar

tmtmtl30 commented:
unit tests are angry but it works locally. my solution for getting ships to spawn on the outpost was a quick hack and i'll think of a better one when i'm not dead tired

UltraBigRat commented:
overall pretty good, some feedback:
Maybe add ship names (and maybe classes) to the  elevator floor buttons since all floors other then the first one are a hangar for a single ship.
The elevator button on the outside should probably just call instead of moving it up or down, and maybe the elevator should only close upon it beginning to move not when people enter. 
overall pretty good though, just needs polish.

tmtmtl30 committed:
minor map fixes

tmtmtl30 committed:
multiple maps / skin, better admin spawning

tmtmtl30 commented:
minor update: admins can now spawn ships in locations other than the outpost

spockye reviewed (dismissed):
![image](https://user-images.githubusercontent.com/79304582/223762489-efa009d2-cff4-435c-864d-50afa86304d1.png)
small decal error here. Try changing the layers around!

![image](https://user-images.githubusercontent.com/79304582/223758772-5561a698-ec55-4c8c-87b6-ceffc1f464dd.png)
did you mean for them to be this faded?

![image](https://user-images.githubusercontent.com/79304582/223759183-a9a012a0-ed0d-4095-9d1e-b898a47c0686.png)
WOAH HIGH TECH HOLOGRAPHIC COMPUTER SCREEN

![image](https://user-images.githubusercontent.com/79304582/223759403-550af2d4-5356-4399-9955-222ccfa7799a.png)
use the table version of the wideband!

![image](https://user-images.githubusercontent.com/79304582/223760204-6fb2e6b8-ffd9-41e2-acb9-34a70b843cd7.png)
Button doesn't work

![image](https://user-images.githubusercontent.com/79304582/223760476-2bcf148e-3c6b-462e-8a39-987d8d5152a5.png)
This button doesn't work

![image](https://user-images.githubusercontent.com/79304582/223760790-9dfef936-e681-4474-bca7-1d0fb7c36e9d.png)
This button doesn't work

goober3 commented:
I don't know if this is a Mapping Issue or what but. a lot of Air Supply Pipes in the outpost are invisible in-game. odd!

goober3 commented:
will there bean option for ships to spawn in Space on ship configs? might be useful for more hazardous ships, like the Metis.

tmtmtl30 commented:
i don't have that planned but it'd definitely be doable code-wise

Erikafox committed:
bjarl base outpost map update

PRing to forks is fun!!!!

tmtmtl30 committed:
adds second outpost map, courtesy spockye

tmtmtl30 committed:
Merge remote-tracking branch 'origin/master' into hangar

tmtmtl30 commented:
spockye has graced the PR with a second map for testing -- the map will be picked randomly at roundstart

BarteG44 commented:
the requirement for a kitchen was a chef holopad. the holopad is here but kitchen is not. i will resort to violence

Erikafox committed:
smaller outpost_test_1

* Update outpost_test_1.dmm

* Update outpost_test_1.dmm

* Update outpost_test_1.dmm

tmtmtl30 committed:
fixes small elevator quirks; outpost map selection

tmtmtl30 committed:
Merge remote-tracking branch 'origin/master' into hangar

tmtmtl30 committed:
Merge branch 'master' into hangar

tmtmtl30 committed:
Merge remote-tracking branch 'origin/master' into hangar

tmtmtl30 committed:
fixes hangars

tmtmtl30 committed:
hopefully fixes outpost forcing

BarteG44 commented:
Add moar holopadd got daym

spockye committed:
outpost hangars

(all except the largest)

spockye committed:
Merge branch 'hangar' of https://github.com/tmtmtl30/Shiptest into pr/1789

tmtmtl30 committed:
removes SSminor_mapping
it was placing mice, satchels, etc on outposts. i consider it deprecated

tmtmtl30 committed:
does a bunch of stuff
- shuffles outpost map template files around, alters structure
- adds ships that spawn in space by default via option in JSON (undoced)
- adds hangar numbers, radio announcement when landing
- adds "fake" elevator levels (untested) for no-level outposts
- some other misc elevator code quality shit
- some misc ship code quality shit
- adds vox to outpost name pool

thgvr commented:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

tmtmtl30 committed:
lots of changes
- fixes some runtimes
- repaths + converts some maps
- minor elevator changes
- fixes hangar numbering
- adds some documentation
- adds Spawn Outpost verb
- removes deprecated code
- adds space_spawn to JSON schema / example

tmtmtl30 committed:
Merge remote-tracking branch 'github/hangar' into hangar

tmtmtl30 committed:
adds unit test for outposts + docking

tmtmtl30 committed:
Merge remote-tracking branch 'origin/master' into hangar

tmtmtl30 committed:
fixes icon files

tmtmtl30 committed:
runs prettier on elevator buttons

tmtmtl30 committed:
fixes linters
(yes, i used text2path. i know what i'm doing)

tmtmtl30 committed:
fixes outpost maps a little

tmtmtl30 committed:
Merge remote-tracking branch 'origin/master' into hangar

tmtmtl30 committed:
adds numbers to all hangars, NOTELEPORT outposts

tmtmtl30 committed:
fixes test2 outpost icon state

tmtmtl30 committed:
Merge remote-tracking branch 'origin/master' into hangar

tmtmtl30 committed:
Merge remote-tracking branch 'origin/master' into hangar

tmtmtl30 committed:
fixes ships hanging around in outpost contents

MarkSuckerberg commented on code/modules/overmap/objects/outpost/elevator/elevator_master.dm:
```diff
@@ -0,0 +1,368 @@
+	/// List of floor datums that the elevator may move between.
+	var/list/datum/floor/floor_list = list()
+
+	/// The index into the floor list at which the
```
the

MarkSuckerberg commented on html/changelogs/AutoChangeLog-pr-1812.yml:
```diff
@@ -0,0 +1,4 @@
+author: MemedHams
+delete-after: true
+changes:
+  - bugfix: The rigger-c json is now correctly pathed to a .dmm file
```
oop

MarkSuckerberg reviewed (changes requested):
just two small non-functional things, otherwise it's legit from what I can tell

tmtmtl30 committed:
addresses review

MarkSuckerberg reviewed (approved):
lgtm

goober3 commented:
no freakin way lois

BarteG44 commented:
At last
